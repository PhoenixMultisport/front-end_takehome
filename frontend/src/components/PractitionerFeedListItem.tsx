import React from 'react'

interface PractitionerFeedListItemParams { itemKey: string, resourceType: string, email: string, url: string }

const PractitionerFeedListItem: React.FC<PractitionerFeedListItemParams> = (props: PractitionerFeedListItemParams): JSX.Element => {
  return (
    <li key={props.itemKey} className="text-left mb-5 bg-amber-50 text-black p-3 rounded-2xl px-10 drop-shadow-md">
      <h1 className="text-xl font-bold">{props.resourceType}</h1>
      <a className="hover:text-amber-300 block text-blue-600" href={`mailto:${props.email}`}>{ props.email || ''}</a>
      <a className="hover:text-amber-300 block text-blue-600" href={props.url}>Website</a>
    </li>
  )
};

export default PractitionerFeedListItem;
