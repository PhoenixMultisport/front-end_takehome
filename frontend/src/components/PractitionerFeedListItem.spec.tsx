import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import PractitionerFeedListItem from './PractitionerFeedListItem';
import { describe } from '@jest/globals';

let documentBody: RenderResult;

describe('<PractitionerFeedListItem />', () => {
  beforeEach(() => {
    documentBody = render(
      <PractitionerFeedListItem
        itemKey={'item-key'}
        resourceType={'resource-type'}
        email={'email'}
        url={'url'}
      />
    );
  });

  it('renders without crashing', () => {
    expect(documentBody.getByText('resource-type')).toBeInTheDocument();
    expect(documentBody.getByText('Website')).toBeInTheDocument();
    expect(documentBody.getByText('Email')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
})
