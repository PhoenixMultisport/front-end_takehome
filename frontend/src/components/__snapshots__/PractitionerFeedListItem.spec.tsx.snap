// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PractitionerFeedListItem /> matches snapshot 1`] = `
<body>
  <div>
    <li
      class="text-left mb-5 bg-amber-50 text-black p-3 rounded-2xl px-10 drop-shadow-md"
    >
      <h1
        class="text-xl font-bold"
      >
        resource-type
      </h1>
      <a
        class="hover:text-amber-300 block text-blue-600"
        href="url"
      >
        Website
      </a>
      <a
        class="hover:text-amber-300 block text-blue-600"
        href="mailto:email"
      >
        Email
      </a>
    </li>
  </div>
</body>
`;
